src/KeysManager.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\pubkey_encrypt\KeysManager.
+ */
+
+namespace Drupal\pubkey_encrypt;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\user\PrivateTempStoreFactory;
+use Drupal\user\UserInterface;
+
+/**
+ * Handles users' Public/Private key pairs.
+ */
+class KeysManager {
+  protected $entityTypeManager;
+  protected $tempStore;
+
+  /*
+   * Constructor with dependencies injected to it.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, PrivateTempStoreFactory $tempStoreFactory) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->tempStore = $tempStoreFactory->get('pubkey_encrypt');
+  }
+
+  /*
+   * Initialize all users' keys.
+   */
+  public function initializeAllUserKeys() {
+    $users = $this->entityTypeManager->getStorage('user')->loadMultiple();
+
+    foreach ($users as $user) {
+      $this->initializeUserKeys($user);
+    }
+  }
+
+  /*
+   * Initialize a specific user's keys.
+   */
+  public function initializeUserKeys(UserInterface $user) {
+    // Generate a Public/Private key pair.
+    $config = array(
+      "config" => "C:/xampp/apache/bin/openssl.cnf",
+      "digest_alg" => "sha512",
+      "private_key_bits" => 4096,
+      "private_key_type" => OPENSSL_KEYTYPE_RSA,
+    );
+    // Create the private and public key.
+    $res = openssl_pkey_new($config);
```

Comment:
So, all of this code logic should be in a second plugin system to generate keys.

In short, here's a brief description of the two plugins...

1. Key Data plugin --- example "password" plugin, returns user passwords
2. Key Generation plugin --- example "openssl" plugin, returns generated key from the Key Data plugin